### PR TITLE
Switch to react-colorful color picker

### PR DESCRIPTION
**Pull request by foxymiles**
**User-Facing Changes**
This replaces the color picker widget with a new widget.

**Description**
This swaps out the fluent color picker with the react-colorful picker: https://github.com/omgovich/react-colorful#customization

Screen Shot 2022-08-12 at 7 19 32 AM

Fixes https://github.com/foxglove/studio/issues/3984

### CODE REVIEW

File: packages/studio-base/package.json [hunk 1]
```diff
@@ -157,6 +158,7 @@
     "natsort": "2.0.3",
     "nearley": "2.20.1",
     "nearley-loader": "2.0.0",
+    "npm": "8.17.0",
```
Comment (2metres) [inline packages/studio-base/package.json:161]: Huh?

File: packages/studio-base/src/components/SettingsTreeEditor/inputs/ColorPickerControl.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+import Stack from "@foxglove/studio-base/components/Stack";
+
+const useStyles = makeStyles()((theme) => ({
+  container: {
+    padding: theme.spacing(2),
+  },
+  picker: {
+    "&.react-colorful": {
+      width: "100%",
```
Comment (2metres) [inline packages/studio-base/src/components/SettingsTreeEditor/inputs/ColorPickerControl.tsx:20]:
```suggestion
      width: "100%",
      gap: theme.spacing(0.5),
```

File: packages/studio-base/src/components/SettingsTreeEditor/inputs/ColorPickerControl.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      ) : (
+        <HexColorPicker
+          className={classes.picker}
+          color={swatchColor}
+          onChange={(newValue) => updateColor(newValue)}
+        />
+      )}
+      <TextField
+        label="Hex"
+        error={editedValueIsInvalid}
```
Comment (2metres) [inline packages/studio-base/src/components/SettingsTreeEditor/inputs/ColorPickerControl.tsx:91]:
```suggestion
        error={editedValueIsInvalid}
        size="small"
```

File: packages/studio-base/src/components/SettingsTreeEditor/inputs/ColorPickerControl.tsx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        InputProps={{
+          startAdornment: <InputAdornment position="start">#</InputAdornment>,
```
Comment (2metres) [inline packages/studio-base/src/components/SettingsTreeEditor/inputs/ColorPickerControl.tsx:93]:
```suggestion
          startAdornment: <TagIcon size="small" />,
```

**Review by 2metres — COMMENTED** [review]
LGTM, a couple of minor tweaks as suggestions

File: packages/studio-base/package.json (hunk 1 above)
Comment (foxymiles) [reply packages/studio-base/package.json:161]: That was a typo. Removed.

**Comment by jtbandes** [comment]
Do we not have any stories in storybook that actually show the picker itself?

**Comment by jtbandes** [comment]
image
Is it intentional that the color swatch got less square?

**Review by jtbandes — APPROVED** [review]

**Comment by foxymiles** [comment]
> Do we not have any stories in storybook that actually show the picker itself?

No and I think we probably should add some that show it and also test entering values directly. It would probably make sense to add those as part of this PR.

**Comment by foxymiles** [comment]
> image
> 
> Is it intentional that the color swatch got less square?

It looks like we were actually specifying it as non-square before but it wasn't being rendered that way for some reason:

```typescript
export const ColorSwatch = muiStyled("div", {
  shouldForwardProp: (prop) => prop !== "color",
})<{ color: string }>(({ theme, color }) => ({
  backgroundColor: color,
  height: theme.spacing(2.5),
  width: theme.spacing(3),
  margin: theme.spacing(0.625),
  borderRadius: 1,
  border: `1px solid ${theme.palette.getContrastText(color)}`,
}));
```

I think it would probably look better square though.

**Comment by foxymiles** [comment]
> image
> 
> Is it intentional that the color swatch got less square?

I changed this to be a square.

**Comment by foxymiles** [comment]
> Do we not have any stories in storybook that actually show the picker itself?

I added some stories for the new color picker input. Good call.

**Pull request merged**